core/src/consensus.rs

```diff
@@ -261,6 +270,7 @@ impl Tower {
             }
 
             if key == *node_pubkey || vote_state.node_pubkey == *node_pubkey {
+                my_latest_landed_vote = vote_state.nth_recent_vote(0).map(|v| v.slot).unwrap_or(0);
```

Comment:
Since nodes can switch vote accounts, this should take the max: `my_latest_landed_vote = my_latest_landed_vote.max(...)`

Reply:
Switched to check by vote account key instead of reverse indexing by node pubkey